src/main/java/ui/components/ButtonThemeSwitch.java

```diff
@@ -0,0 +1,55 @@
+package main.java.ui.components;
+
+import com.formdev.flatlaf.FlatDarkLaf;
+import com.formdev.flatlaf.FlatLaf;
+import com.formdev.flatlaf.FlatLightLaf;
+import com.formdev.flatlaf.extras.FlatAnimatedLafChange;
+import org.kordamp.ikonli.fontawesome5.FontAwesomeSolid;
+import org.kordamp.ikonli.swing.FontIcon;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class ButtonThemeSwitch extends JButton {
```

Comment:
I also think that each class should have a short JavaDoc explaining its inteded function. Even when the name makes it clear.